static/js/containers/DashboardPage.js

```diff
@@ -119,14 +111,16 @@ class DashboardPage extends React.Component {
     dispatch:                 Dispatch,
     setCalculatorVisibility:  (b: boolean) => void,
     ui:                       UIState,
-    courseTeamEmail:          EmailState,
+    email:                    AllEmailsState,
     documents:                DocumentsState,
     fetchDashboard:           () => void,
     orderReceipt:             OrderReceiptState,
     courseEnrollments:        CourseEnrollmentsState,
     financialAid:             FinancialAidState,
     location:                 Object,
     pearson:                  PearsonAPIState,
+    emailDialog:              React$Element<*>|null,
```

Comment:
on a side note, I think it would be better to have the HOC render the dialog component, just so the wrapped component doesn't have to worry about it.